lib/spack/spack/cmd/ci.py

```diff
@@ -816,6 +835,167 @@ def _gitlab_artifacts_url(url: str) -> str:
     return urlunparse(parsed._replace(path="/".join(parts), fragment="", query=""))
 
 
+def validate_standard_versions(pkg, versions: VersionList) -> bool:
```

Comment:
I think a function like this could be useful in `package_base.py`, but this isn't really what you'd want there. This function is largely for iterating through the versions and printing the specific outputs of this command.

Reply:
Agreed. But could be refactored to process the versions and let the caller generate the outputs (as would be expected of a command)?

Reply:
In playing around with "unit testing" this I was surprised to find `tty.die()` in `stage.py` (among others) as I distinctly remember being told that it should not be used in modules at that level; rather Spack-specific exceptions should be raised that the caller (command) can handle as needed.